staging/src/k8s.io/apiserver/pkg/audit/request_test.go

```diff
@@ -226,3 +227,44 @@ func TestCopyWithoutManagedFields(t *testing.T) {
 		})
 	}
 }
+
+func TestEncodeDeleteOptions(t *testing.T) {
+
+	policy := metav1.DeletePropagationBackground
+
+	tests := []struct {
+		name       string
+		object     runtime.Object
+		gv         schema.GroupVersion
+		serializer runtime.NegotiatedSerializer
+	}{
+		{
+			name: "encode unversioned DeleteOptions",
+			object: &metav1.DeleteOptions{
+				PropagationPolicy: &policy,
+			},
+			gv:         metav1.SchemeGroupVersion,
+			serializer: metainternalversionscheme.Codecs,
+		},
+		{
+			name: "encode v1 DeleteOptions",
```

Comment:
I don't understand the difference between those two cases (TypeMeta shouldn't really matter here...)

Reply:
ok..., I have a wrong understanding of Codec here